runtime/sema/type.go

```diff
@@ -669,6 +674,10 @@ func (t *OptionalType) IsResourceType() bool {
 	return t.Type.IsResourceType()
 }
 
+func (t *OptionalType) IsPrimitiveType() bool {
+	return false
```

Comment:
We might want to also consider optional types which wrap a primitive type to in turn be primitive, too

Reply:
Done - https://github.com/onflow/cadence/pull/2962/commits/61e79f1c1b35bfc2e029b43f597e5916d6ce738a.